tools/tpm2_pcrallocate.c

```diff
@@ -0,0 +1,241 @@
+                             ESYS_TR_PASSWORD, ESYS_TR_NONE, ESYS_TR_NONE,
+                             &ctx.pcrSelection, &allocationSuccess,
+                             &maxPCR, &sizeNeeded, &sizeAvailable);
+    if (rval != TSS2_RC_SUCCESS) {
+        LOG_ERR("Could not allocate PCRs.");
+        LOG_PERR(Esys_PCR_Allocate, rval);
+        return false;
+    }
+
+    if (!allocationSuccess) {
+        LOG_ERR("Allocation failed. Size Needed: %i, Size available: %i",
+                sizeNeeded, sizeAvailable);
+        return false;
+    }
+    (void)(maxPCR);
+
+    return true;
+}
+
+static bool on_arg(int argc, char** argv){
+    int i;
+    uint32_t pcr;
+
+    for (i = 0; i < argc; i++){
+        if (tpm2_util_string_to_uint32(argv[i], &pcr)) {
+            if (ctx.pcrSelection.count == 0) {
+                LOG_ERR("Got PCR Index without Bank selection");
+                return false;
+            }
+
+            if (pcr >= TPM2_MAX_PCRS) {
+                LOG_ERR("Got out of bound PCR Index: \"%s\"", argv[i]);
+                return false;
+            }
+
+            if (pcr > 24)
+                ctx.pcrSelection.pcrSelections[ctx.pcrSelection.count - 1].
+                    sizeofSelect = 4;
+
+            ctx.pcrSelection.pcrSelections[ctx.pcrSelection.count - 1].
+                pcrSelect[pcr / 8] |= 1 << (pcr % 8);
+
+            continue;
+        }
+
+        if (argv[i][strlen(argv[i])-1] == ':') {
+            char *opt = strdup(argv[i]);
+            opt[strlen(opt) - 1] = '\0';
+            TPM2_ALG_ID alg = tpm2_alg_util_from_optarg(opt, tpm2_alg_util_flags_hash);
+            if (alg == TPM2_ALG_ERROR) {
+                LOG_ERR("Unknowdn hash alg");
+                return false;
+            }
+            ctx.pcrSelection.pcrSelections[ctx.pcrSelection.count++].hash = alg;
+            ctx.pcrSelection.pcrSelections[ctx.pcrSelection.count - 1].
+                sizeofSelect = 3;
+            continue;
+        }
+
+        if (!strcmp(argv[i], "all")) {
+            ctx.pcrSelection.pcrSelections[ctx.pcrSelection.count - 1].
+                pcrSelect[0] = 0xff;
+            ctx.pcrSelection.pcrSelections[ctx.pcrSelection.count - 1].
+                pcrSelect[1] = 0xff;
+            ctx.pcrSelection.pcrSelections[ctx.pcrSelection.count - 1].
+                pcrSelect[2] = 0xff;
+            continue;
+        }
+
+        LOG_ERR("Unknown parameter");
+        return false;
+    }
+
+    return true;
+}
+
+static bool on_option(char key, char *value) {
+    switch (key) {
+    case 'P':
+        ctx.platauth.size = strlen(value);
+        if (ctx.platauth.size > sizeof(ctx.platauth.buffer)) {
+            LOG_ERR("Password too long");
+            ctx.platauth.size = 0;
+            return false;
+        }
+        memcpy(&ctx.platauth.buffer, value, ctx.platauth.size);;
+        break;
+    }
+
+    return true;
+}
+
+bool tpm2_tool_onstart(tpm2_options **opts) {
+    memset(&ctx, 0, sizeof(ctx));
```

Comment:
Remove that `memset` if ctx is made `static`

Reply:
Done